src/index.js

```diff
@@ -77,46 +82,28 @@ class DelegatedContentRouting {
 
     options.maxTimeout = options.maxTimeout || DEFAULT_MAX_TIMEOUT
 
-    this.dht.findProvs(key.toString(), {
-      timeout: `${options.maxTimeout}ms` // The api requires specification of the time unit (s/ms)
-    }, callback)
+    this._httpQueue.add(() =>
+      this.dht.findProvs(key.toString(), {
+        timeout: `${options.maxTimeout}ms` // The api requires specification of the time unit (s/ms)
+      }, callback)
+    ).catch(callback)
```

Comment:
I don't know exactly how `promisify-es6` works under the hood, does it call callback _as well as_ return a promise? I would have thought it did one or the other.

Maybe we can make this a bit easier to understand as well as eliminate any doubt?

```js
    this._httpQueue.add(() =>
      this.dht.findProvs(key.toString(), {
        timeout: `${options.maxTimeout}ms` // The api requires specification of the time unit (s/ms)
      })
    ).then(res => callback(null, res), callback)
```

Reply:
if callback is present, `promisify-es6` will call function as-is; if not, it will return a Promise that injects callback as the last argument before calling the thing

[promisify-es6/index.js#L31-L32](https://github.com/manuel-di-iorio/promisify-es6/blob/3fb20250c335aebf22775e6b0fbb42e3f6eb662c/index.js#L31-L41)

Should work the same, so I've switched to your version.